src/X.PagedList/BasePagedList.cs

```diff
@@ -20,16 +20,23 @@ namespace X.PagedList;
 [PublicAPI]
 public abstract class BasePagedList<T> : IPagedList<T>
 {
-    protected readonly List<T> Subset = new();
-
     public const int DefaultPageSize = 100;
 
+    private readonly List<T> _subset = new();
+
     /// <summary>
     /// Parameterless constructor.
     /// </summary>
     protected internal BasePagedList()
     {
     }
+    
+    protected void FillSubset(IEnumerable<T> set)
+    {
+        _subset.Clear();
+        
+        _subset.AddRange(set);
```

Comment:
Just concerning the implementation, I would prefer just

`_subset = set.ToList();`
or
`_subset = new List<T>(set);`

instead of the two other commands. Garbage collector will throw away the old list, which is much less expensive than actively clearing it. For lists that are not huge, this is micro-optimization though.